src/components/common/ColonyDecisions/DecisionPreview/DecisionPreview.tsx

```diff
@@ -32,8 +33,13 @@ const DecisionPreview = () => {
   const { user, walletConnecting, userLoading } = useAppContext();
   const walletAddress = user?.walletAddress || '';
 
+  const [decisionPreview, setDecisionPreview] = useState<Decision>();
   const decision = getDecisionFromLocalStorage(walletAddress);
 
+  if (!decisionPreview && decision) {
+    setDecisionPreview(decision);
+  }
```

Comment:
I always thought you're supposed to only set state inside callbacks or `useEffect`. While this doesn't trigger render loop because of the condition, I think it might be better to put this inside a `useEffect` - see https://stackoverflow.com/a/74036012

Reply:
@jakubcolony I change this in a subsequent pr to use redux so all the decisions components can access the same state. 

However, on the subject of setting state, I'm using a pattern recommended by the React docs: https://beta.reactjs.org/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes

Also enjoyed [this talk](https://www.youtube.com/watch?v=bGzanfKVFeU&t) which is more or less covers what's included in the link above (i.e when to use useEffect and how to use it less).

Reply:
@willm30 Thanks for sharing, looks like I'll have some reading for the weekend!